xtevent/_eventivstatic.ado

```diff
@@ -1,172 +1,255 @@
-				if !_rc loc Floop = e(F)
-				if `Floop' > `Fstart' {
-					loc Fstart = `Floop'
-					loc proxyiv "`v'"				
-				}			
-			}
-			di as text _n "Lead `proxyiv' selected."
-		}
-		
-	}
-	
-	* Parse proxyiv and generate leads if neccesary
-	loc rc=0
-	foreach v in `proxyiv' {
-		cap confirm integer number `v'
-		loc rc = `rc' + _rc
-	}
-	* If numlist take as leads of z
-	if `rc' == 0 {
-		loc insvars ""
-		foreach v in `proxyiv' {
-			qui gen _f`v'`z' = f`v'.`z' if `touse'
-			loc insvars "`insvars' _f`v'z"
-		}
-		loc instype = "numlist"
-	}
-	else {
-		foreach v in `proxyiv' {
-			confirm numeric variable `v'
-		}
-		loc insvars = "`proxyiv'"
-	}
-		
-	* Main regression
-	
-	if "`reghdfe'"=="" {
-		if "`fe'" == "nofe" {
-		loc cmd "ivregress 2sls"
-		loc ffe ""
-		}
-		else {
-			loc cmd "xtivreg"
-			loc ffe "fe"
-		}
-		`cmd' `varlist' (`proxy' = `insvars') `z' `tte' [`weight'`exp'] if `touse' , `ffe' `options'
-	}
-	else {
-		loc cmd "reghdfe"
-		if "`fe'" == "nofe" & "`te'"=="" & "`absorb'"=="" {						
-			loc noabsorb "noabsorb"
-			loc abs ""
-		}
-		else if "`fe'" == "nofe" & "`te'"=="" & "`absorb'"!="" {						
-			loc noabsorb "noabsorb"
-			loc abs "absorb(`absorb')"
-		}
-		else if "`fe'" == "nofe" & "`te'"!=""{
-			loc abs "absorb(`te' `absorb')"	
-			loc noabsorb ""
-		}
-		else if "`fe'" != "nofe" & "`te'"==""{
-			loc abs "absorb(`i' `absorb')"	
-			loc noabsorb ""
-		}
-		else {
-			loc abs "absorb(`i' `te' `absorb')"	
-			loc noabsorb ""
-		}
-		ivreghdfe `varlist' (`proxy' = `insvars') `z' [`weight'`exp'] if `touse', `absorb' `noabsorb' `options'
-	}
-	
-	
-	
-	
-	if "`te'" == "note" loc tte ""
-	else loc tte "i.`t'"
-	
-	mat `bb' = e(b)
-	mat `VV' = e(V)
-	mat `delta'=e(b)
-	mat `Vdelta'=e(V)	
-	
-	* Drop variables
-	
-	if "`instype'"=="numlist" {
-		foreach v in `proxyiv' {
-			drop _f`v'`z' 
-		}
-	}
-	
-	tokenize `varlist'
-	loc depvar "`1'"
-	
-	return matrix b = `bb'
-	return matrix V = `VV'
-	return matrix delta = `delta'
-	return matrix Vdelta = `Vdelta'
-	loc names: subinstr global names ".." " ", all
-	loc names: subinstr local names `"""' "", all
-	return local names = "`names'"	
-	return local cmd = "`cmd'"
-	return local depvar = "`depvar'"
-		
-end
+* _eventivstatic.ado 1.00 Aug 24 2021
+
+version 11.2
+
+cap program drop _eventivstatic
+program define _eventivstatic, rclass
+	#d;
+	syntax varlist(fv ts numeric) [aw fw pw] [if] [in], /* Covariates go in varlist. Can add fv ts later */
+	panelvar(varname) /* Panel variable */
+	timevar(varname) /* Time variable */
+	policyvar(varname) /* Policy variable */	
+	proxy (varlist numeric) /* Proxy variable(s) */
+	[
+	proxyiv(string) /* Instruments. Either numlist with lags or varlist with names of instrumental variables */
+	nofe /* No fixed effects */
+	note /* No time effects */	
+	reghdfe /* Use reghdfe for estimation */
+	absorb(string) /* Absorb additional variables in reghdfe */ 
+	impute(string)
+	STatic

```

Comment:
@Constantino-Carreto-Romero does this command need this `static` option?

Reply:
When splitting _eventgenvars.ado into two sections, one that imputes the policyvar and other that generates the event-time dummies, I used the option `static` to do it: the first part, that imputes the policyvar, is always available for static or no static ados, only depending if they call the `impute` option. On the other hand, the part of _eventgenvars.ado that generates the event-time dummies, is only available if `"static"==""`, that is why when _eventivstatic.ado and _eventolsstatic.ado call _eventgenvars.ado, they do it including the option `static` in their call, so they will not access to the part of _eventgenvars.ado that generates the event-time dummies.